bungiesearch/utils.py

```diff
@@ -80,9 +74,36 @@ def delete_index_item(item, model_name, refresh=True):
             src.get_es_instance().delete(index_name, model_name, item_es_id)
         except NotFoundError as e:
             logging.warning('NotFoundError: could not delete {}.{} from index {}: {}.'.format(model_name, item_es_id, index_name, str(e)))
-        
+
         if refresh:
             src.get_es_instance().indices.refresh(index=index_name)
 
+def create_indexed_document(index_instance, model_items, action):
+    '''
+    Creates the document that will be passed into the bulk index function.
+    Either a list of serialized objects to index, or a a dictionary specifying the primary keys of items to be delete.
+    '''
+    data = []
+    if action == 'delete':
+        for pk in model_items:
+            data.append({'_id': pk, '_op_type': action})
+    else:
+        for doc in model_items:
+            if index_instance.matches_indexing_condition(doc):
+                data.append(index_instance.serialize_object(doc))
+    return data
+
+def filter_model_items(index_instance, model_items, model_name, start_date, end_date):
+    ''' Filters the model items queryset based on start and end date.'''
+    if index_instance.updated_field is None:
+        logging.warning("No updated date field found for {} - not restricting with start and end date".format(model_name))
```

Comment:
This is a logger warning in Haystack instead of a hard error, and I think we should follow suit. The problem with the `ValueError` from before was that it was difficult to apply the update functionality across varied types of model indices. Let me know what you think about this!